docs/extended_api/pipelines/pipeline/consumer_wait.md

```diff
@@ -0,0 +1,62 @@
+---
+grand_parent: Pipelines
+parent: cuda::pipeline
+---
+
+# `cuda::pipeline::consumer_wait*`
+
+```c++
+// Header `<cuda/pipeline>`.
+
+template <cuda::thread_scope Scope>
+__host__ __device__
+void cuda::pipeline<Scope>::consumer_wait();
+```
+
+Blocks the current thread until all operations committed to the current
+  _pipeline stage_ complete
+
+```c++
+// Header `<cuda/pipeline>`.
+
+template <cuda::thread_scope Scope>
+template <typename Rep, typename Period>
+__host__ __device__
+bool cuda::pipeline<Scope>::consumer_wait_for(
+  cuda::std::chrono::duration<Rep, Period> const& duration);
+```
+
+Blocks the current thread until all operations committed to the current
+  _pipeline stage_ complete or after the specified timeout duration
+
+```c++
+// Header `<cuda/pipeline>`.
+
+template <cuda::thread_scope Scope>
+template <typename Clock, typename Duration>
+__host__ __device__
+bool cuda::pipeline<Scope>::consumer_wait_until(
+  cuda::std::chrono::time_point<Clock, Duration> const& time_point);
+```
+
+Blocks the current thread until all operations committed to the current
+  _pipeline stage_ complete or until specified time point has been reached
```

Comment:
Same comment about the split of these into multiple code blocks as on the `memcpy_async` doc file.

This file needs a comment that this functionality is not a part of the github-released open-source library.